sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -736,6 +736,12 @@ object SQLConf {
       .checkValue(weight => weight >= 0 && weight <= 1, "The weight value must be in [0, 1].")
       .createWithDefault(0.7)
 
+  val JOIN_REORDER_DP_STAR_FILTER =
+    buildConf("spark.sql.cbo.joinReorder.dp.star.filter")
+      .doc("Applies star-join filter heuristics to cost based join enumeration.")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
In Spark 2.2, we introduced a couple of new configuration parameters in optimizer area.  In order to play on the safe side, we set the default value to false.  I suggest that we can change the default value to true AFTER we are sure that the new optimizer feature does not cause any regression.  I think the system regression/integration test suites help us make a decision.